components/integrations_marketplace/marketplace_modal.jsx

```diff
@@ -0,0 +1,189 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import PropTypes from 'prop-types';
+import React from 'react';
+import {FormattedMessage} from 'react-intl';
+import {Tabs, Tab} from 'react-bootstrap';
+
+import FullScreenModal from 'components/widgets/modals/full_screen_modal';
+import RootPortal from 'components/root_portal';
+import QuickInput from 'components/quick_input';
+import LocalizedInput from 'components/localized_input/localized_input';
+import PluginIcon from 'components/widgets/icons/plugin_icon.jsx';
+import LoadingScreen from 'components/loading_screen.jsx';
+
+import {t} from 'utils/i18n';
+import {localizeMessage} from 'utils/utils';
+
+import MarketplaceItem from './marketplace_item';
+
+import './marketplace_modal.scss';
+
+const MarketplaceTabs = {
+    ALL_PLUGINS: 'allPlugins',
+    INSTALLED_PLUGINS: 'installed',
+};
+
+export default class MarketplaceModal extends React.Component {
+    static propTypes = {
+        show: PropTypes.bool,
+        installedPlugins: PropTypes.array.isRequired,
+        marketplacePlugins: PropTypes.array.isRequired,
+        actions: PropTypes.shape({
+            closeModal: PropTypes.func.isRequired,
+            getMarketplacePlugins: PropTypes.func.isRequired,
+        }).isRequired,
+    }
+
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            tabKey: MarketplaceTabs.ALL_PLUGINS,
+            loading: true,
+        };
+    }
+
+    componentWillMount() {
+        this.getMarketplacePlugins();
+    }
+
+    getMarketplacePlugins = async () => {
+        await this.props.actions.getMarketplacePlugins();
```

Comment:
what'll happen if this call fails, not sure what is the pattern for catching errors, is that part of the mm-redux action?

Reply:
Correct, the error handling will be done in redux with a specific action that will have error information. WIP